app/src/main/java/ml/docilealligator/infinityforreddit/subreddit/shortcut/ShortcutManager.java

```diff
@@ -0,0 +1,36 @@
+package ml.docilealligator.infinityforreddit.subreddit.shortcut;
+
+import android.content.Context;
+import android.content.Intent;
+import android.graphics.Bitmap;
+
+import androidx.core.content.pm.ShortcutInfoCompat;
+import androidx.core.content.pm.ShortcutManagerCompat;
+import androidx.core.graphics.drawable.IconCompat;
+
+import ml.docilealligator.infinityforreddit.BuildConfig;
+import ml.docilealligator.infinityforreddit.activities.ViewSubredditDetailActivity;
+
+public class ShortcutManager {
+    private static ShortcutInfoCompat getInfo(Context context, String subreddit, Bitmap icon) {
+        final Intent shortcut = new Intent(context, ViewSubredditDetailActivity.class);
+        shortcut.setPackage(context.getPackageName());
+        shortcut.setAction(Intent.ACTION_MAIN);
+        shortcut.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+        shortcut.putExtra(ViewSubredditDetailActivity.EXTRA_SUBREDDIT_NAME_KEY, subreddit);
+
+        String shortcutId = BuildConfig.APPLICATION_ID + ".shortcut." + subreddit;
```

Comment:
What if the user wants to add the same shortcut again? Does this make the previous one on the home screen disappear?

Reply:
No, there will be two of the same shortcuts. I'm just using the system apis, there's no control on what's already on the home screen.